x-pack/plugins/search_indices/public/components/quick_stats/quick_stats.tsx

```diff
@@ -0,0 +1,163 @@
+import { i18n } from '@kbn/i18n';
+import { Mappings } from '../../types';
+import { countVectorBasedTypesFromMappings } from './mappings_convertor';
+import { QuickStat } from './quick_stat';
+
+export interface QuickStatsProps {
+  index: Index;
+  mappings: Mappings;
+}
+
+export const SetupAISearchButton: React.FC = () => {
+  return (
+    <EuiPanel hasBorder={false} hasShadow={false} color="transparent">
+      <EuiFlexGroup gutterSize="s" direction="column" alignItems="center">
+        <EuiFlexItem grow={false}>
+          <EuiText>
+            <h6>
+              {i18n.translate('xpack.searchIndices.quickStats.setup_ai_search_description', {
+                defaultMessage: 'Use Elastic to build powerful AI Search',
+              })}
+            </h6>
+          </EuiText>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButton
+            href="https://www.elastic.co/guide/en/elasticsearch/reference/current/semantic-search.html"
+            target="_blank"
+            data-test-subj="setupAISearchButton"
+          >
+            {i18n.translate('xpack.searchIndices.quickStats.setup_ai_search_button', {
+              defaultMessage: 'Setup now',
+            })}
+          </EuiButton>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiPanel>
+  );
+};
+
+export const QuickStats: React.FC<QuickStatsProps> = ({ index, mappings }) => {
+  const [open, setOpen] = React.useState<boolean>(false);
+  const { euiTheme } = useEuiTheme();
+  const mappingStats = countVectorBasedTypesFromMappings(mappings);
+  const vectorFieldCount =
+    mappingStats.sparse_vector + mappingStats.dense_vector + mappingStats.semantic_text;
+
+  return (
+    <EuiPanel
+      paddingSize="none"
+      data-test-subj="quickStats"
+      hasShadow={false}
+      css={() => ({
+        border: euiTheme.border.thin,
+        background: euiTheme.colors.lightestShade,
+        overflow: 'hidden',
+      })}
+    >
+      <EuiFlexGroup gutterSize="none">
+        <EuiFlexItem>
+          <QuickStat
+            open={open}
+            setOpen={setOpen}
+            icon="documents"
+            iconColor="black"
+            title={i18n.translate('xpack.searchIndices.quickStats.document_count_heading', {
+              defaultMessage: 'Document count',
+            })}
+            data-test-subj="QuickStatsDocumentCount"
+            secondaryTitle={<EuiI18nNumber value={index.documents ?? 0} />}
+            stats={[
+              {
+                title: i18n.translate('xpack.searchIndices.quickStats.documents.totalTitle', {
+                  defaultMessage: 'Total',
+                }),
+                description: <EuiI18nNumber value={index.documents ?? 0} />,
+              },
+              {
+                title: i18n.translate('xpack.searchIndices.quickStats.documents.indexSize', {
+                  defaultMessage: 'Index Size',
+                }),
+                description: index.size ?? 0,
```

Comment:
is this size already i18n'd curious why we used `<EuiI18nNumber>`  for the other stats but not this one?

Reply:
yeh size is already localised by the API. It should always return a value (with no documents it returns '0b'. I've updated the fallback to match however.

Reply:
updated